src/main/java/net/earthcomputer/clientcommands/command/KitCommand.java

```diff
@@ -151,6 +161,18 @@ private static int list(ServerCommandSource source) {
         return kits.size();
     }
 
+    private static int preview(ServerCommandSource source, String name) throws CommandSyntaxException {
+        ListTag kit = kits.get(name);
+        if (kit == null) {
+            throw NOT_FOUND_EXCEPTION.create(name);
+        }
+
+        PlayerInventory tempInv = new PlayerInventory(client.player);
+        tempInv.deserialize(kit);
+        client.send(() -> client.openScreen(new PreviewScreen(new PlayerScreenHandler(tempInv, true, client.player), tempInv, name)));
```

Comment:
Is there any reason the opening of the screen has to be deferred? Why not just call `client.openScreen` here directly?
Also, is using a player screen handler likely to have any side effects?

Reply:
It has to be. In your Discord I posted an explanation for why this has to be like this. I would link to it but I’m on my phone at the moment. Will get back to you

Reply:
So, there reason is this:
After executing a command, the current screen will be closed (the chat hud). But if you open a new screen in a command, that new screen will be closed instantly along with the chat hud. To _fix_ this issue some people registered a certain event on the client's tick:
```java
ClientTickEvents.START_CLIENT_TICK.register(client -> {
    if (KitCommand.openScreen != null) {
        client.openScreen(KitCommand.openScreen);
        KitCommand.openScreen = null;
    }
});
```I kind of hate this solution to be honest. Using `client.send()` is a bit cleaner, in my opinion.

In regards to instantiating a `new PlayerScreenHandler`, I don't know - sorry. It was either this or creating an entire `ClientPlayerEntity`, which is probably worse. Another option would be to create an entirely new `Screen`, but this suffices for now, I guess.

Reply:
Who the heck suggested to use the tick event to fix that? Nope, you've done it the right way, nice one :+1: 
You should add a comment to explain why this workaround is needed though, so someone reading the code will know.

Reply:
Everyone whom I asked suggested this 🤣. I refused to use that.
- [OroArmor](https://discord.com/channels/507304429255393322/507982478276034570/829625729213792286)
- [Random Forge people apparently](https://discord.com/channels/703363189081112647/703598704266969099/830223532109332570)
I'll add a comment.